pkg/operator/ceph/object/controller.go

```diff
@@ -319,3 +338,36 @@ func updateStatus(client client.Client, name types.NamespacedName, status string
 	}
 	logger.Debugf("object store %q status updated to %q", name, status)
 }
+
+func (r *ReconcileCephObjectStore) waitForObjectBucketCleanup(objectstore *cephv1.CephObjectStore) error {
+	for retryCount := 0; retryCount < objectstoreDeleteMaxRetries; retryCount++ {
+
+		selector := fmt.Sprintf("metadata.namespace=%s", objectstore.Namespace)
+		objectBuckets, err := r.bktclient.ObjectbucketV1alpha1().ObjectBuckets().List(metav1.ListOptions{FieldSelector: selector})
+		if err != nil {
+			return errors.Wrapf(err, "failed to get buckets for objectstore %q in namespace %q", objectstore.Name, objectstore.Namespace)
+		}
+		if len(objectBuckets.Items) == 0 {
+			logger.Infof("no objectbuckets for objectstore %q to clean up.", objectstore.Namespace)
+			break
+		}
+
+		retryCount++
+		if retryCount == objectstoreDeleteMaxRetries {
+			bucketNames := make([]string, 0)
+			for _, bucket := range objectBuckets.Items {
+				bucketNames = append(bucketNames, bucket.Name)
+			}
```

Comment:
@leseb : May be it is very naive question, if I understand correctly here object store is getting deleted. And we trying fetch info from obcs to see buckets is getting deleted. IMHO obc/ob will still present even if objectstore is deleted.
```console
# kubectl delete -f cluster/examples/kubernetes/ceph/object-test-sam.yaml 
cephobjectstore.ceph.rook.io "my-store" deleted

# kubectl get obc
NAME                  AGE
ceph-delete-bucket    3m46s
ceph-delete-bucket2   85s

# kubectl get ob
NAME                              AGE
obc-default-ceph-delete-bucket    3m52s
obc-default-ceph-delete-bucket2   92s
```

Reply:
Yes, it will but its existence does not make much sense anymore. The buckets are not accessible anymore and the data are gone.

It's just not clear from that code where the deletion of OBC is happening though...

Reply:
There is a check in the yaml file - "preservePoolsOnDelete". When this is false, rook deletes the Object Pools [here](https://github.com/rook/rook/blob/1ea73ce1233fcaa4a1b5518967eef2dcbd0cade6/pkg/operator/ceph/object/objectstore.go#L77). And deleting the pools would delete the object buckets, is what I read [here](https://github.com/rook/rook/pull/4683#issuecomment-575315487). 
So we will only wait for the object buckets to be deleted when:
1. "preservePoolsOnDelete" is set to false. ( That's in this PR)
2.  when the cleanup policy is set for the deletion of Object buckets. (Planned for another PR)

Reply:
Let me rephrase my concern, I agree that if you delete the pool, all the (actual)buckets will be removed but obc's still may exists.  And here for checking, the bucket name is fetched from OBC, which was confusing for me.

Reply:
I was wrong in my comment [above](https://github.com/rook/rook/pull/5436#discussion_r423681260).  Object buckets are still there after the pools are deleted (like you mentioned [here](https://github.com/rook/rook/pull/5436#discussion_r423658160)). So current changes won't work. I'm working on fixing that now.